src/question/order_select/mod.rs

```diff
@@ -0,0 +1,272 @@
+mod builder;
+
+use std::io;
+
+use ui::{
+    backend::Backend,
+    events::EventIterator,
+    style::Color,
+    widgets::{self, List, Text},
+    Prompt, Widget,
+};
+
+use crate::{Answer, Answers, ListItem};
+
+use super::{
+    handler::{Filter, Transform, Validate},
+    Choice,
+};
+
+pub use builder::OrderSelectBuilder;
+
+#[cfg(test)]
+mod tests;
+
+// =============================================================================
+//
+// =============================================================================
+
+#[derive(Debug, Default)]
+pub(super) struct OrderSelect<'a> {
+    choices: super::ChoiceList<Text<String>>,
+    max_number_len: usize,
+    order: Vec<usize>,
+    moving: bool,
+
+    transform: Transform<'a, [ListItem]>,
+    validate: Validate<'a, [usize]>,
+    filter: Filter<'a, Vec<usize>>,
+}
+
+impl widgets::List for OrderSelect<'_> {
+    fn render_item<B: ui::backend::Backend>(
+        &mut self,
+        index: usize,
+        hovered: bool,
+        mut layout: ui::layout::Layout,
+        b: &mut B,
+    ) -> std::io::Result<()> {
+        let symbol_set = ui::symbols::current();
+
+        if hovered {
+            if self.moving {
+                b.set_bg(Color::Cyan)?;
+                b.set_fg(Color::Black)?;
+            } else {
+                b.set_fg(Color::Cyan)?;
+            }
+
+            write!(b, "{} ", symbol_set.pointer)?;
+        } else {
+            b.write_all(b"  ")?;
+        }
+
+        layout.offset_x += 4;
+
+        let mut index_str = (index + 1).to_string();
+        index_str.extend(
+            ' '.to_string()
+                .repeat(self.max_number_len - index_str.len())
+                .chars()
+        );
+
+        write!(b, "{} ", index_str)?;
```

Comment:
I think it's better to extends this code to RawSelect as well, because it make the view feel more consistent

The main problem with the current approach is that the options are not aligned properly and it's more noticable when the list can loops with a lot of options

Reply:
I suggest instead to implement this for RawSelect and other modules that use indices too, because it will make the display more consistent, especially when the list can loops with a lot of options

I'm pretty sure i didn't implement the algorithm in the most efficient way though

Reply:
You make a fair point. I will update the visuals of `RawSelect` to match yours with two small changes. I have added a dot right after the number and made it right aligned. You can replace lines [64-75](https://github.com/Lutetium-Vanadium/requestty/blob/cb0293d46025f4350c47fb8b8e01d3df5d9f7d41/src/question/order_select/mod.rs#L64-L75) with the following code.
```rust
write!(
    b,
    "  {:>width$}. ",
    index + 1,
    width = self.max_number_len
)?;

layout.offset_x += self.max_index_width + 4;
```

This uses the in-built [padding functionality](https://doc.rust-lang.org/std/fmt/index.html#fillalignment) of `std::fmt`.

I also just noticed that in [`height_at`](https://github.com/Lutetium-Vanadium/requestty/blob/cb0293d46025f4350c47fb8b8e01d3df5d9f7d41/src/question/order_select/mod.rs#L95-L98) you add 4 directly to `offset_x`, it should instead add `self.max_number_len as u16 + 4`.

Reply:
Oh, i wasn't aware of this functionality. Thanks for let me know about this.

Also, i suggest to remove the spaces before `{:>width}` in the `write!` as it makes the space between the pointer and the index too large.
It will make a too big difference with `MultiSelect`.

Reply:
Yes, please remove them. I had copy pasted what I wrote for `RawSelect`, which doesn't have the pointer and so those spaces weren't rendered before